core/base/scalarFieldSmoother/ScalarFieldSmoother.h

```diff
@@ -47,14 +47,19 @@ namespace ttk {
       // Pre-condition functions.
       if(triangulation) {
         triangulation->preconditionVertexNeighbors();
+#if TTK_ENABLE_MPI
+        if(ttk::isRunningWithMPI())
+          triangulation->preconditionDistributedVertices();
```

Comment:
please remove this 4-liner and add a call `preconditionDistributedVertices()` at the end of the function `setInputPoints` and `setStellarInputPoints` of `core/base/triangulation`.

(that way, it is always called by default if MPI is detected).

Reply:
Hmmm, `preconditionDistributedVertices()` needs the GlobalIdArray which is set in `MPIPreconditioning()`in `ttkAlgorithm.cpp`. Would it be more elegant to move the `MPIPreconditioning` into the triangulation itself, or to move the preconditioning into ttkAlgorithm?

Reply:
another option would be to add a call to `preconditionDistributedVertices()` right at the end of the implementation of `setGlobalIdsArray()`. I think that would make more sense. what do you think?